platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/pool/TransactionPoolNexus.java

```diff
@@ -148,19 +152,35 @@ public synchronized boolean submitApplicationTransaction(@NonNull final SwirldTr
             return false;
         }
 
-        if (transaction.getSize() > maximumTransactionSize) {
+        int transactionSize = transactionConversion(transaction).getSize();
+        if (transactionSize > maximumTransactionSize) {
             // FUTURE WORK: This really should throw, but to avoid changing existing API this will be changed later.
             illegalTransactionLogger.error(
                     EXCEPTION.getMarker(),
                     "transaction has {} bytes, maximum permissible transaction size is {}",
-                    transaction.getSize(),
+                    transactionSize,
                     maximumTransactionSize);
             return false;
         }
 
         return submitTransaction(transaction, false);
     }
 
+    /**
+     * Convert a transaction to a {@link ConsensusTransactionImpl}. This is a hack because {@code OneOf<>} interface
+     * can not return a size of the transaction.
+     *
+     * @param transaction the transaction to convert
+     * @return the converted transaction
+     */
+    private ConsensusTransactionImpl transactionConversion(final OneOf<PayloadOneOfType> transaction) {
```

Comment:
for this I think we need a standalone utility that returns the transaction size. we will need it elsewhere. also, the utility should not use the wrappers, it should calculate the size the same way the wrapper would. it would also be good to add a unit test to see if the size always matches the wrapped size.